File: spark/01_test/WordCount.scala
import org.apache.spark.{SparkConf, SparkContext}

object WordCount {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("WordCount")
    val sc = new SparkContext(conf)
    val localPath = "D:/scala/bigData/01_test/src/main/resources"

    val rdd = sc.textFile(localPath + "/WordCount_The_Man_of_Property.txt")
    val data = rdd.flatMap(_.split(" ")) // "_"表示输入
      .map((_, 1)) // "()"表示一个tuple--"(word,1)"
      .reduceByKey(_ + _) // 一个value加上另一个value，此时输出--"(word,count)"
      .map {
        x => x._1 + "\t" + x._2 // 输出“word count”
      }

    data.saveAsTextFile(localPath + "/wordCount_out")

  }
}

File: spark/03_pyspark_test/wordcount.py
#!/usr/local/bin/python
# --coding:utf-8--

from pyspark import SparkContext, SparkConf

# 本地测试
input_path = "./"
output_path = "./"
# 集群测试
# input_path = ""
# output_path = ""


def sp(x):
    return x.strip().split(" ")


if __name__ == "__main__":
    conf = SparkConf().setMaster("local").setAppName("word_count")
    sc = SparkContext(conf=conf)

    in_file = sc.textFile(input_path + "/The_Man_of_Property.txt")
    rdd = in_file \
        .flatMap(sp) \
        .map(lambda word: (word, 1)) \
        .reduceByKey(lambda a, b: a + b) \
        .map(lambda x: " ".join([x[0], str(x[1])])) \
        .saveAsTextFile(output_path + "/wordCount_out")

    # result = rdd.collect()
    #
    # for line in result:
    #     print(line)
    sc.stop()
